internal/store/store.go

```diff
@@ -463,7 +463,7 @@ func (s *Store) PushCompose(composeID uuid.UUID, bp *blueprint.Blueprint, checks
 		repos = append(repos, source.RepoConfig())
 	}
 
-	pipeline, err := s.distro.Pipeline(bp, repos, checksums, arch, composeType)
+	pipeline, err := s.distro.Pipeline(bp, repos, checksums, arch, composeType, size)
```

Comment:
I think you should change this to pass 0 in the commit that changes the interface, then update in this commit (introducing the size parameter to PushCompose()) to pass on the real size.